src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -422,19 +407,20 @@ public void updateOdometry() {
         m_poseEstimator.addVisionMeasurement(
           estimatedPose.estimatedPose.toPose2d(),
           estimatedPose.timestampSeconds,
-          m_isOnChargeStation ? VisionConstants.kChargeStationVisionPoseStdDevs : VisionConstants.kBaseVisionPoseStdDevs.plus(
+          m_chargeStationVision ? VisionConstants.kChargeStationVisionPoseStdDevs : VisionConstants.kBaseVisionPoseStdDevs.plus(
             currentEstimatedPoseTranslation.getDistance(closestTagPoseTranslation) * VisionConstants.kVisionPoseStdDevFactor
           )
         );
       }
+      
+      // If it used vision after going over the charge station, it should trust vision normally again
+      if(estimatedPoses.size()>0){
+        m_chargeStationVision = false;
+      }
```

Comment:
Do you need to set it false only if there are vision targets present?
In fact, why is this a member variable anymore? You don't use it anywhere else right? You could just make it a local boolean

Reply:
I thought about that, but I want it to use vision 1 time after the angle is over that value. If it can't see an April tag while it is on the charge station, I want it to keep this variable as true until it can, so it corrects the pose after the charge station.

Reply:
Ok I see. Maybe make that more clear in comments.